src/simularium/VisData.ts

```diff
@@ -98,91 +51,62 @@ class VisData {
         if (util.ThreadUtil.browserSupportsWebWorkers()) {
             this.setupWebWorker();
         }
-        this.frameCache = [];
-        this.frameDataCache = [];
-        this.cacheFrame = -1;
-        this.enableCache = true;
+        this.currentFrameNumber = -1;
+        this.frameCache = new VisDataCache();
         this.frameToWaitFor = 0;
         this.lockedForFrame = false;
         this.timeStepSize = 0;
+
+        this.onError = noop;
     }
 
-    //get time() { return this.cacheFrame < this.frameDataCache.length ? this.frameDataCache[this.cacheFrame] : -1 }
-    public get currentFrameData(): FrameData {
-        if (this.frameDataCache.length > 0) {
-            if (this.cacheFrame < 0) {
-                return this.frameDataCache[0];
-            } else if (this.cacheFrame >= this.frameDataCache.length) {
-                return this.frameDataCache[this.frameDataCache.length - 1];
-            } else {
-                return this.frameDataCache[this.cacheFrame];
-            }
-        }
+    public setOnErrorCallback(callback: (error: FrontEndError) => void): void {
+        this.onError = callback;
+    }
```

Comment:
We are now passing down the `on.Error` prop to VisData and we do in `VisGeometry`, mostly to catch the edge case where an incoming frame is larger than the maxSize of the cache.

I'm curious if I should do the same and pass this error handling all the way down into `VisDataCache` as well to catch frame access errors, or alternately if there is a different approach to be using here.